test/FastTests/Client/Queries/QueryTests.cs

```diff
@@ -32,6 +33,134 @@ private class B
             public byte Byte { get; set; }
         }
 
+        private class Article
+        {
+            public string Title { get; set; }
+            public string Description { get; set; }
+            public bool IsDeleted { get; set; }
+        }
+        
+        [Fact]
+        public async Task Query_CreateClausesForQueryDynamicallyWhenTheQueryEmpty()
+        {
+            using (var store = GetDocumentStore())
+            {
+                const string id1 = "users/1";
+                const string id2 = "users/2";
+                
+                using (var session = store.OpenSession())
+                {
+                    session.Store(new Article
+                        {
+                            Title = "foo",
+                            Description = "bar",
+                            IsDeleted = false
+                            
+                        },
+                        id1);
+                    
+                    session.Store(new Article
+                        {
+                            Title = "foo",
+                            Description = "bar",
+                            IsDeleted = true
+                        },
+                        id2);
+                    session.SaveChanges();
+                }
+                
+                using (var session = store.OpenAsyncSession())
+                {
+                    var query = await session.Advanced.AsyncDocumentQuery<Article>()
```

Comment:
There is no assertion in this test.

Do 2 things:
1. ToString() and assert if generated RQL query is correct
2. ToListAsync and check if returned results are correct